### PR TITLE
cleancode:util.go

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
cleancode, the modification point make code clean and light

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot ok to test

File: federation/pkg/kubefed/util/util.go [hunk 1]
```diff
@@ -254,9 +254,9 @@ func GetVersionedClientForRBACOrFail(hostFactory cmdutil.Factory) (client.Interf
 				}
 				return hostFactory.ClientSetForVersion(&gv)
 			}
-			for i := 0; i < len(g.Versions); i++ {
-				if g.Versions[i].GroupVersion != "" {
-					gv, err := schema.ParseGroupVersion(g.Versions[i].GroupVersion)
+			for _, version := range g.Versions {
+				if version.GroupVersion != "" {
+					gv, err := schema.ParseGroupVersion(version.GroupVersion)
```
Comment (resouer) [inline federation/pkg/kubefed/util/util.go]: Sorry to say this ... but I don't think this make too much difference. Do you have reason to do this, for example, code style best practice?

**Comment by resouer** [comment]
@k8s-bot ok to test

File: federation/pkg/kubefed/util/util.go (hunk 1 above)
Comment (zhangxiaoyu-zidif) [inline federation/pkg/kubefed/util/util.go]: In function the original point equals the modification point. However modification is much cleaner and lighter than the original point. First, it does not need len(xxx) to know the loop count, and the index is not necessary for the function. Don't you think so?

**Comment by zhangxiaoyu-zidif** [comment]
@k8s-bot verify test this

**Pull request merged**